From 6f417fec0071fa2814d4449cc78db8da8f009d99 Mon Sep 17 00:00:00 2001
From: Hector Martin <marcan@marcan.st>
Date: Tue, 22 Nov 2022 13:52:24 +0900
Subject: [PATCH] quirks: Add Apple MTP touchpad quirk

Apple M2 (and presumably newer) laptops now embed the touchpad
controller into the main SoC, and use a new internal communications
protocol between it and the main CPU. This isn't really a "bus" like
SPI or I2C, so the downstream kernel driver currently uses the (not
well supported) HOST bus type. MatchBus can't match on that, so let's
just use a name match (plus the vendor ID, which is still valid and
the usual Apple one).

Signed-off-by: Hector Martin <marcan@marcan.st>
---
 quirks/50-system-apple.quirks | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/quirks/50-system-apple.quirks b/quirks/50-system-apple.quirks
index 7f7a81cd36d5..a6d6eea91d5a 100644
--- a/quirks/50-system-apple.quirks
+++ b/quirks/50-system-apple.quirks
@@ -105,6 +105,15 @@ AttrSizeHint=104x75
 AttrTouchSizeRange=150:130
 AttrPalmSizeThreshold=1600
 
+[Apple Laptop Touchpad (MTP)]
+MatchUdevType=touchpad
+MatchName=Apple*MTP*
+MatchVendor=0x5AC
+ModelAppleTouchpad=1
+AttrSizeHint=104x75
+AttrTouchSizeRange=150:130
+AttrPalmSizeThreshold=1600
+
 # The Linux applespi driver currently uses the Synaptics vendor for some reason
 [Apple Laptop Touchpad (SPI)]
 MatchUdevType=touchpad
-- 
2.38.1

